### ISSUE TITLE
Driver seems to inhibit shutdown?

### ISSUE THREAD

**Issue by steeley** [bondagit/aes67-linux-daemon#62]
on my RPI I'm loading the driver with insmod from /etc/rc.local. Works fine, but
when I want to shutdown or reboot the system hangs waiting for the ethernet port to become free.

Could this be the driver holding on to it?
If so, is there a better  way to load the driver ? I need it to load before pulseaudio.
( maybe it can go in /lib/modules/5.10.63-v8+/kernel/sound/drivers ?)

The only other thing associated with the ethernet port is ptp4l, and the aes67-daemon

**Comment by bondagit** [comment]
The driver cannot be unloaded if it's in use. 
Any player using the RAVENNA ALSA device  and the daemon have to be terminated before attempting to unload the module.

**Comment by steeley** [comment]
I don't seem to have anything using it, but it still blocks reboot/shutdown.

 is there a sensible way to fix this?
system cannot be used if it blocks. other audio drivers do not have this issue.

**Comment by bondagit** [comment]
The driver unloads properly for me.
I suspect that your problem is related to something else.

**Comment by steeley** [comment]
can the driver be loaded automatically at start - /etc/modules perhaps?

**Comment by bondagit** [comment]
yes. If you are using a Linux Ubuntu distro you can achieve this using the followingi nstructions:
1. Install you kernel module with:
```
cd 3rdparty/ravenna-alsa-lkm/driver
make modules_install
```
If this doesn't work because you miss kernel certificates follow the instructions at:[No OpenSSL sign-file signing_key.pem](https://superuser.com/questions/1214116/no-openssl-sign-file-signing-key-pem-leads-to-error-while-loading-kernel-modules)

2. update the module dependencies with:
`sudo depmod -a`

3. add MergingRavennaALSA to /etc/modules

**Comment by djsakisd** [comment]
@bondagit I'm facing the same issue on raspberry 4, console output during reboot/shutdown show this error : 
`unregister_netdevice waiting for eth0 to become free usage count = 3`

Here's a couple of things i tried - every time same issue happens: 

- Tried both 5.10 & 6.6 kernels on debian buster & bookworm (32bit with flag arm_64bit set to 0). (TAG 1.7 of driver due to [this issue ](https://github.com/bondagit/ravenna-alsa-lkm/issues/19#issuecomment-2145630435)
- Tried ubuntu 22 (can't remember architect) with default shipped kernel - same problem.
- Tried to stop aes daemon service before rebooting.
- Tried to remove sink before rebooting.
- Load driver is loaded during boot 
- Loaded driver manually with lsmod.
- Removed sink from webui / stop daemon & reboot.

When reboot/shutdown **works**: 
1. When there is **no sink defined** in webui, reboot/shutdown 
2. On Ubuntu 24.04 with default kernel ( i think its 6.8).

If there is anything i cant test to help resolving this let me know.
